results/cmd/api/main_test.go

```diff
@@ -437,3 +443,133 @@ func TestDeleteTaskRun(t *testing.T) {
 		t.Fatalf("same taskrun not supposed to be deleted again: %v", r.String())
 	}
 }
+func TestListTaskRuns(t *testing.T) {
+	// Create a temporary database
+	srv, err := setupTestDB(t)
+	if err != nil {
+		t.Fatalf("failed to setup db: %v", err)
+	}
+	ctx := context.Background()
+
+	// Create a bunch of taskruns for testing
+	t1 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "31415926",
+			Name:      "taskrun",
+			Namespace: "default",
+		},
+	}
+	t2 := &pb.TaskRun{
+		ApiVersion: "v1alpha1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "43245243",
+			Name:      "task",
+			Namespace: "default",
+		},
+	}
+	t3 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "1234556",
+			Name:      "mytaskrun",
+			Namespace: "demo",
+		},
+	}
+	t4 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "demo",
+		},
+	}
+	t5 := &pb.TaskRun{
+		ApiVersion: "v1alpha1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "official",
+		},
+	}
+	taskruns := []*pb.TaskRun{t1, t2, t3, t4, t5}
+	for _, ts := range taskruns {
+		if _, err := srv.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{
+			TaskRun: ts,
+		}); err != nil {
+			t.Fatalf("could not create taskrun: %v", err)
+		}
+	}
+	tt := []struct {
+		name   string
+		filter string
+		expect []*pb.TaskRun
+	}{
+		{
+			name:   "test simple query",
+			filter: `taskrun.api_version=="v1beta1"`,
+			expect: []*pb.TaskRun{t1, t3, t4},
+		},
+		{
+			name:   "test query with simple function",
+			filter: `taskrun.metadata.name.endsWith("run")`,
+			expect: []*pb.TaskRun{t1, t3, t4, t5},
+		},
+		{
+			name:   "test empty filter",
+			filter: "",
+			expect: taskruns,
+		},
+		{
+			name:   "test invalid field",
+			filter: `task.name=="newtaskrun"`,
+		},
+		{
+			name:   "test invalid value type with no double quotes around value",
+			filter: `taskrun.name==newtaskrun`,
+		},
+		{
+			name:   "test value not existing in the server record",
+			filter: `taskrun.metadata.name=="notaskrun"`,
+			expect: nil,
+		},
+		{
+			name:   "test invalid field outside of our defined top level field",
+			filter: `taskrun.unexistfield=="notaskrun"`,
+		},
+		{
+			name:   "test valid field but not boolean experssion",
+			filter: `taskrun.api_version"`,
+		},
+		{
+			name:   "test random word input",
+			filter: `tekton`,
+		},
+		{
+			name:   "test if field is case sensitive",
+			filter: `taskrun.MetaData.name=="notaskrun"`,
+		},
+	}
+	// Test if we can find inserted taskruns
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			res, err := srv.ListTaskRunsResult(ctx, &pb.ListTaskRunsRequest{
+				Filter: tc.filter,
+			})
+			if err != nil && status.Code(err) != codes.InvalidArgument {
```

Comment:
We should be more explicit here and check for the returned status. It's difficult to determine whether something is passing because they returned no values vs they returned InvalidArgument.

Reply:
done